metrics/opentelemetry/metrics.go

```
@@ -0,0 +1,158 @@
+package opentelemetry
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/gobeam/stringy"
+	"github.com/golang/glog"
+	"go.opentelemetry.io/otel/metric"
+)
+
+type (
+	// Elem represents a metric element.
+	Elem struct {
+		Name  string
+		Value reflect.Value
+		Tag   reflect.StructTag
+	}
+)
+
+// FindAllMetrics finds all metrics in the given struct.
+func FindAllMetrics(m any, prefix string) <-chan Elem {
+	ret := make(chan Elem, 16)
+
+	go func() {
+		defer close(ret)
```

Each goroutine should has `recovery` because panic will crash the process

---

This is only invoked at initialization time and is not calling any client-provided code… at runtime a fail fast approach seems reasonable - what would recover do but exit?